src/google_takeout_metadata/processor_batch.py

```
@@ -45,44 +45,107 @@ def process_batch(batch: List[Tuple[Path, Path, List[str]]], clean_sidecars: boo
+            for media_path, _, _ in batch:
+                is_image = media_path.suffix.lower() in IMAGE_EXTS  
+                stats.add_processed_file(media_path, is_image)
+            return len(batch)
+        elif "character(s) could not be encoded" in stderr_msg:
+            error_type = "encoding_error"
+            error_msg = "Problème d'encodage de caractères (émojis, accents)"
+            logger.warning(f"⚠️ {error_msg}. Détails: {stderr_msg.strip()}")
+        else:
+            error_type = "exiftool_error"
+            error_msg = f"Erreur exiftool (code {exc.returncode}): {stderr_msg.strip() or 'Erreur inconnue'}"
+            logger.error(f"❌ Échec du lot de {len(batch)} fichier(s). {error_msg}")
```

Ruff (0.12.2)

Use `logging.exception` instead of `logging.error`

Replace with `exception`

(TRY400)